app/assets/javascripts/user.js

```diff
@@ -7,30 +7,66 @@ $(function () {
   })
 
   .done(function(data){
-    var rivals = data[0];
-    var wins = data[1];
-    var mostPlayedNames = data[2];
-    var winsWith = data[3];
-    $("#rivals").highcharts({
-        chart: {
-            type: "bar"
-        },
-        title: {
-            text: "Top Rivals"
-        },
-        xAxis: {
-            categories: rivals
-        },
-        yAxis: {
+    var opponent_names = data[0];
+    var wins_against = data[1];
+    var losses_against = data[2];
+    var mostPlayedNames = data[3];
+    var winsWith = data[4];
+
+    $('#rivals').highcharts({
+            chart: {
+                type: 'bar'
+            },
             title: {
-                text: "Losses to each:"
-            }
-        },
-        series: [{
-            name: "Rivals",
-            data: wins
-        }]
-    });
+                text: 'Wins and Losses by Opponent'
+            },
+            xAxis: [{
+                categories: opponent_names,
+                reversed: false,
+                labels: {
+                    step: 1
+                }
+            }, { // mirror axis on right side
+                opposite: true,
+                reversed: false,
+                categories: opponent_names,
+                linkedTo: 0,
+                labels: {
+                    step: 1
+                }
+            }],
+            yAxis: {
+                title: {
+                    text: null
+                },
+                labels: {
+                    formatter: function () {
+                        return Math.abs(this.value);
+                    }
+                }
+            },
+
+            plotOptions: {
+                series: {
+                    stacking: 'normal'
+                }
+            },
+
+            tooltip: {
+                formatter: function () {
+                    return '<b>' + this.series.name + ' vs. ' + this.point.category +
+                    ': ' + Highcharts.numberFormat(Math.abs(this.point.y), 0) +'</b><br/>';
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>'Highcharts' is not defined.